tests/test_gp.py

```diff
@@ -0,0 +1,193 @@
+
+import sys
+sys.path.append('../')
+
+import unittest
```

Comment:
Perhaps if it's too much effort we can keep the tests in unittest as opposed to switching to pytest as they're just tests, any thoughts @InfProbSciX and @leojklarner ?

Reply:
Yeah - I'll standardise - are we using pytest?